glean-core/src/upload/directory.rs

```diff
@@ -140,14 +200,19 @@ mod test {
 
     #[test]
     fn test_doesnt_panic_if_no_pending_pings_directory() {
+        // Create a new directory_manager
         let dir = tempfile::tempdir().unwrap();
-        assert!(process_pings_dir(&dir.path()).is_err());
+        let tmpname = dir.path().display().to_string();
+        let directory_manager = PingDirectoryManager::new(&tmpname);
```

Comment:
He, if the manager takes a `PathBuf` we don't even need to go through the display implementation.